src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2687,11 +2690,66 @@ void PgScrubber::log_cluster_warning(const std::string& warning) const
   m_osds->clog->do_log(CLOG_WARN, warning);
 }
 
-ostream& PgScrubber::show(ostream& out) const
+
+ostream& PgScrubber::show_concise(ostream& out) const
 {
-  return out << " [ " << m_pg_id << ": " << m_flags << " ] ";
+  /*
+  * 'show_concise()' is only used when calling operator<< thru the ScrubPgIF,
+  * i.e. only by the PG when creating a standard log entry.
+  *
+  * desired outcome (only relevant for Primaries):
+  *
+  * if scrubbing:
+  *   (urgency flags)
+  *   or (if blocked)
+  *   (*blocked*,urgency flags)
+  *
+  * if not scrubbing:
+  *   either nothing (if only periodic scrubs are scheduled)
+  *   or [next scrub: effective-lvl, urgency, rpr,
+  */
+  if (!is_primary()) {
+    return out;
+  }
+
+  if (m_active) {
+    const auto flags_txt = get_cur_scrub_flags_text();
```

Comment:
Given all of the other work done in this function, is it *really* worth caching this?  It's debug information, so doing it this way means that any time we want to rely on it, we'll have to audit to be certain that it really got updated correctly.

Reply:
(I am assuming you refer to the work performed by show_concise())

I think it is. For one thing - this is called for every log line. And if getting
it wrong - it is, as you say, only a helper during debug (and the correct data
is available to see immediately, if we suspect the code).
But it's a good question, so I've benchmarked the effect the
caching would have on performance. The results are
(on my laptop, w/ just the relevant functions):

If no "interesting" bits are set in the flags (the common case):
```
Not cached - median        339 ns
Cached -   median          272 ns (80% of 339)

```
If there is text to output for the flags:
```
Not cached - median        578 ns
Cached -   median          353 ns  (61% of 578)

```
I consider this meaningful enough to include the cache.